query/control/controller.go

```diff
@@ -87,23 +100,53 @@ type Config struct {
 	ExecutorDependencies []flux.Dependency
 }
 
+// complete will fill in the defaults, validate the configuration, and
+// return the new Config.
+func (c *Config) complete() (Config, error) {
+	config := *c
+	if config.InitialMemoryBytesQuotaPerQuery == 0 {
+		config.InitialMemoryBytesQuotaPerQuery = config.MemoryBytesQuotaPerQuery
+	}
+	if config.MaxMemoryBytes == 0 {
+		config.MaxMemoryBytes = int64(config.ConcurrencyQuota) * config.MemoryBytesQuotaPerQuery
+	}
+
+	if err := config.Validate(); err != nil {
+		return Config{}, err
+	}
+	return config, nil
+}
+
 func (c *Config) Validate() error {
 	if c.ConcurrencyQuota <= 0 {
 		return errors.New("ConcurrencyQuota must be positive")
 	}
 	if c.MemoryBytesQuotaPerQuery <= 0 {
 		return errors.New("MemoryBytesQuotaPerQuery must be positive")
 	}
+	if c.InitialMemoryBytesQuotaPerQuery <= 0 {
+		return errors.New("InitialMemoryBytesQuotaPerQuery must be positive")
+	}
+	if c.MaxMemoryBytes <= 0 {
```

Comment:
Looks like there may be an inconsistency between line 130 and line 139? They are checking the same value.

Reply:
Fixed. It seems like I accidentally made this same check twice.